### PR TITLE
Addin

### PR DESCRIPTION
**Pull request by mserrano626**
A little sloppy.  I don't know all the cool stuff JS can do yet.

### CODE REVIEW

**Comment by ghost** [comment]
Cool, is this for the displaying the parent nodes?

**Comment by mserrano626** [comment]
Yeah. It displays the parents

**Comment by ghost** [comment]
also looks like you accidentally committed a `.swp` file. you can just do 
`git rm .pascal.y.swp` and then commit that

**Comment by mserrano626** [comment]
my bad.  I'll fix that

File: pascal.y [hunk 1]
```diff
@@ -172,6 +172,27 @@
         }
       })
     };
+
+    var addIn = function () {
+      for(var j = 0; j < blocks.length ; j++ ){
```
Comment (ghost) [inline pascal.y:177]: you could do this in the `addBlock` function by just saying `block.in = []`, but if you dont know what i mean dont sweat it

File: pascal.y [hunk 2]
```diff
@@ -172,6 +172,27 @@
...
+        blocks[j].in = [];	
+      }
+			
+      for(var j = 0; j < blocks.length ; j++ ){
+        for(var prop in blocks[j].out){
```
Comment (ghost) [inline pascal.y:182]: prop was kinda confusing, maybe rename to `id`

File: pascal.y [hunk 3]
```diff
@@ -172,6 +172,27 @@
...
+          var vout = blocks[j].out[prop];
+          var outId = blocks[j].id;
+				
+          for(var i = 0; i < blocks.length; i++){
+            if(blocks[i].id == vout){
+              blocks[i].in.push(outId);
+
```
Comment (ghost) [inline pascal.y:189]: extra space sorry :S

File: pascal.y [hunk 4]
```diff
@@ -172,6 +172,27 @@
...
+          for(var i = 0; i < blocks.length; i++){
+            if(blocks[i].id == vout){
+              blocks[i].in.push(outId);
+	
+            }
+          }	
+        }			
+      }
+			
+		};
```
Comment (ghost) [inline pascal.y:195]: the indentation on this is a little off

File: pascal.y [hunk 5]
```diff
@@ -549,7 +571,6 @@ if_statement:
   IF boolean_expression THEN statement ELSE statement {
     $2.type = 'if';
     $2.out = [];
-
```
Comment (ghost) [inline pascal.y:552]: probably accidentally deleted this, its kinda nice to have the line

**Comment by ghost** [comment]
a few tiny things, you dont have to fix them if you dont want, i can always just fix them real quick later. other than that nice job dude for your first time with js! :+1: 

im out for the night, merge whenever idc

**Pull request merged**